### PR TITLE
dplyr fixes

### PR DESCRIPTION
**Pull request by onurfiliz**
The upcoming release of dplyr has a few changes that we need to adapt to:

- Default implementation for `db_query_fields` now uses `dbFetch(result, 0)`. We use the previous `dbListFields` form instead
- `src_translate_env` gets renamed to `sql_translate_env`
- Allow passing `Inf` to `dbFetch`

### CODE REVIEW

File: R/db_query_fields.R [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# of patent rights can be found in the PATENTS file in the same directory.
+
+#' S3 implementation of \code{sql_translate_env} for Presto.
+#'
+#' @rdname dplyr_function_implementations
+#' @keywords internal
+#' @export
+db_query_fields.PrestoConnection <- function(con, sql, ...) {
+  fields <- dplyr::build_sql(
+    "SELECT * FROM ", sql, " WHERE 0=1",
```
Comment (cosinequanon) [inline R/db_query_fields.R]: Instead of `WHERE 0=1` can we do `LIMIT 0`? It is more readable imo

**Comment by cosinequanon** [comment]
Is this all backward compatible? At least until the last stable version of dplyr?

**Comment by onurfiliz** [comment]
Yes, it should be. I am planning to submit to CRAN right after the merge, so I'm making sure that it is indeed backwards compatible.

**Comment by cosinequanon** [comment]
Sweet, lgtm

**Pull request merged**